src/options/normalizeOptions.js

```diff
@@ -203,6 +201,33 @@ function normalizeProjection (options) {
   else return `EPSG:${projection}`
 }
 
+/**
+ * Ensure idField is set if metadata doesn't have a value but a field named OBJECTID is present
+ * @param {object} metadata
+ */
+function normalizeIdField (options, features = []) {
+  const collection = options.collection || {}
+  const metadata = collection.metadata || {}
+  const feature = features[0] || {}
+  const featureProperties = feature.properties || feature.attributes || {}
+  let idField = null
+
+  // First, check metadata for idField
+  if (metadata.idField) idField = metadata.idField
+
+  // Check metadata.fields for and OBJECTID property
+  else if (_.find(metadata.fields, { name: 'OBJECTID' })) idField = 'OBJECTID'
+  // Check features for an OBJECTID property
+  else if (features.length > 0 && featureProperties.OBJECTID) idField = 'OBJECTID'
+
+  // If there are features, check that the idField is one of the properties
+  if (process.env.NODE_ENV !== 'production' && process.env.KOOP_WARNINGS !== 'suppress' && features.length > 0 && !featureProperties[idField]) {
```

Comment:
If `metadata.idField` is `null`, we want `idField` returned as null; this is because we use `idField===null` to determine if we need a OBJECTID to be created for the dataset (see [here](https://github.com/koopjs/winnow/blob/04fbad615751b0371cf3a4fc88fd9b5b5e36d105/src/sql.js#L142)).

I've added a test to show that if If `metadata.idField` is `null`, the function returns null.  https://github.com/koopjs/winnow/pull/104/commits/fd9f8924c6c6f83ee210d962ca88b13f80b4d4ef

Reply:
What I mean is the following warning mentions `requested provider has "idField" assignment`, but the `if` condition does not check whether the `idField` is actually not empty.

Reply:
Ok, I understand.  Thanks, excellent catch.  Updated in 96da3b3.